test_combat.py

```diff
@@ -0,0 +1,45 @@
+from unittest.mock import Mock
+
+import pytest
+
+import enemy_catalog
+import entities
+import game
+import helper
+import items
+from ansi_tags import ansiprint
+from definitions import CombatTier
+
+
+def replacement_clear_screen():
+    '''Replacement for game.view.clear() so that I can see the test output'''
+    print("\n--------------------------\n")
+
+@pytest.mark.only
+def test_new_combat_basic(monkeypatch):
+    entities.random.seed(123)
+    all_cards = [c() for c in items.cards]
+
+    # Create player
+    player = entities.Player(health=100, block=0, max_energy=3, deck=all_cards)
+
+    # Create bad guys
+    acidslime = enemy_catalog.AcidSlimeS()
+    jawworm = enemy_catalog.JawWorm()
+
+    # Create combat object
+    combat = game.Combat(player, CombatTier.NORMAL, [acidslime, jawworm])
+
+    game_map = Mock()
+
+    # Patch the input
+    responses = iter("11\n11\n2\n")
```

Comment:
What's this?

Reply:
[iter is a built-in function that returns an iterable](https://www.programiz.com/python-programming/methods/built-in/iter) (like a generator expression). It's essentially the same as:

```
for c in "11\n11\n2\n"
  yield c
```

This allows it to be called repeatedly. We patch `input()` with it, so it gets called various times and we return a single character. So "11\n" is "play the first card, select the first target, press enter to continue". We do that twice. And then "2\n" is "play the 2nd card", which is a Defend card. This allows you to script the game inputs.